src/Experiment/src/AssemblyBuilder.cs

```diff
@@ -7,21 +7,69 @@
 using System.Text;
 using System.Threading.Tasks;
 using System.Reflection;
+using System.Reflection.Metadata.Ecma335;
+using System.Reflection.Metadata;
+using System.IO;
+using System.Reflection.PortableExecutable;
+using System.Runtime.CompilerServices;
 
 namespace System.Reflection.Emit.Experimental
 {
+
     public class AssemblyBuilder: System.Reflection.Assembly
     {
-        internal AssemblyBuilder() { }
+        private static readonly Guid _s_guid = new Guid("87D4DBE1-1143-4FAD-AAB3-1001F92068E6");//Some random ID, Need to look into how these should be generated.
+        private static readonly BlobContentId _s_contentId = new BlobContentId(_s_guid, 0x04030201);
+        private BlobBuilder _emptyBlob = new BlobBuilder();
+        public override string? FullName { get; }
+        private static MetadataBuilder _metadata = new MetadataBuilder();
+        private static IDictionary<string,ModuleBuilder> moduleStorage = new Dictionary<string, ModuleBuilder>();
+        public AssemblyBuilder() { }
+        private AssemblyBuilder(string name) 
+        {
+            FullName = name;
+        }
 
         public void Save(string assemblyFileName)
         {
-            //New method
+            if (assemblyFileName == null)
+            {
+                throw new ArgumentNullException();
+            }
+            if (FullName == null)
+            {
+                throw new ArgumentNullException();
+            }
+            //Add assembly metadata
+            _metadata.AddAssembly(//Metadata is added for the new assembly - Current design - metdata generated only when Save method is called.
+               _metadata.GetOrAddString(value: FullName),//FullName, CultureName?
+               version: new Version(1, 0, 0, 0),
```

Comment:
version, culture and publicKey should come from the AssemblyName that was used to create the assembly.

Reply:
@MosheWolberg Looks like this feedback hasn't been addressed yet.